### ISSUE TITLE
aXe / accessibility: missing region around Skip link in theme-classic

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [facebook/docusaurus#6252]
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Prerequisites

- [X] I'm using the latest version of Docusaurus.
- [X] I have tried the `npm run clear` or `yarn clear` command.
- [X] I have tried `rm -rf node_modules yarn.lock package-lock.json` and re-installing packages.
- [X] I have tried creating a repro with https://new.docusaurus.io.
- [X] I have read the console error message carefully (if applicable).

### Description

[aXe 4.1: region](https://dequeuniversity.com/rules/axe/4.1/region) specifies:

> Ensure all content is contained within a landmark region, designated with HTML5 landmark elements and/or ARIA landmark regions.
> 
> ...
> 
> Navigating a web page is far simpler for screen reader users if the content splits between multiple high-level sections. Content outside of sections is difficult to find, and the content's purpose may be unclear.

The _Skip to main content link_ is an excellent + necessary thing to have on a webpage (♥) but is itself not in any region right now. That's causing aXe to log a failure when run with Enable Best Practices enabled.

### Steps to reproduce

1. Create a new Docusaurus site with the classic theme, such as today's https://typescript-eslint.io
2. Run aXe on it such as with the [aXe browser extensions](https://www.deque.com/axe/browser-extensions) and Settings > Enable Best Practices enabled

Alternately, see these [cypress-axe](https://www.npmjs.com/package/cypress-axe) failures: https://github.com/typescript-eslint/typescript-eslint/runs/4685834360?check_suite_focus=true

```json
"html": "<div><a href=\"#\" class=\"skipToContent__-_-node_modules-@docusaurus-theme-classic-lib-next-theme-SkipToContent-styles-module\">Skip to main content</a></div>",
"target": [
    "#__docusaurus > div:nth-child(2)"
],
"failureSummary": "Fix any of the following:\n  Some page content is not contained by landmarks"
```

### Expected behavior

Some kind of role wrapped around the skip-to-main-content link.

For example, on codecademy.com/docs, we use a `role="region"`.

### Actual behavior

No region around the link.

### Your environment

- Public source code: https://github.com/typescript-eslint/typescript-eslint/pull/4362
- Public site URL: https://typescript-eslint.io
- Docusaurus version used: `2.0.0-beta.14`
- Environment name and version (e.g. Chrome 89, Node.js 16.4): N/A
- Operating system and version (e.g. Ubuntu 20.04.2 LTS): N/A

### Reproducible demo

https://github.com/typescript-eslint/typescript-eslint/pull/4362

### Self-service

- [X] I'd be willing to fix this bug myself.

**Comment by Josh-Cena** [comment]
Makes sense! Feel free to send a PR😉

**Comment by lex111** [comment]
@Josh-Cena probably did not find appropriate landmark for that element (the choice of `nav` was wrong), and did not know about `region` role, so this is good fix :+1:
